tests/test_lfs.py

```diff
@@ -74,3 +85,197 @@ def test_lfs(tmp_dir: TmpDir, scm: Git, lfs_objects: TmpDir):
+):
+    client = LFSClient.from_git_url("http://git.example.com/namespace/project.git")
+    recorder = CallbackResultRecorder()
+
+    with aioresponses() as m:
+        m.post(
+            f"{client.url}/objects/batch",
+            callback=recorder.record(
+                CallbackResult(
+                    status=HTTPStatus.TOO_MANY_REQUESTS,
+                    headers=rate_limit_header(),
+                    reason="Too many requests",
+                )
+            ),
+            repeat=True,
+        )
+
+        with pytest.raises(ClientResponseError, match="Too many requests"):
+            client.download(storage, [Pointer(oid=FOO_OID, size=FOO_SIZE)])
+
+        results = recorder[f"{client.url}/objects/batch"]
+        assert [r.status for r in results] == [429] * 5
+
+
+@pytest.mark.parametrize(
+    "rate_limit_header",
+    [
+        lambda: {"Retry-After": "1"},
+        lambda: {"RateLimit-Reset": f"{int(time()) + 1}"},
+        lambda: {"X-RateLimit-Reset": f"{int(time()) + 1}"},
+    ],
+)
+def test_rate_limit_max_retries_objects(
+    storage: LFSStorage, rate_limit_header: Callable[[], dict[str, str]]
+):
+    client = LFSClient.from_git_url("http://git.example.com/namespace/project.git")
+    lfs_batch_url = f"{client.url}/objects/batch"
+    lfs_object_url = f"http://git-lfs.example.com/{FOO_OID}"
+    recorder = CallbackResultRecorder()
+
+    with aioresponses() as m:
+        m.post(
+            lfs_batch_url,
+            callback=recorder.record(
+                CallbackResult(
+                    status=HTTPStatus.OK,
+                    headers={"Content-Type": "application/vnd.git-lfs+json"},
+                    payload={
+                        "transfer": "basic",
+                        "objects": [
+                            {
+                                "oid": FOO_OID,
+                                "size": FOO_SIZE,
+                                "authenticated": True,
+                                "actions": {
+                                    "download": {
+                                        "href": lfs_object_url,
+                                    }
+                                },
+                            }
+                        ],
+                        "hash_algo": "sha256",
+                    },
+                )
+            ),
+        )
+        m.get(
+            lfs_object_url,
+            callback=recorder.record(
+                CallbackResult(
+                    status=HTTPStatus.TOO_MANY_REQUESTS,
+                    headers=rate_limit_header(),
+                    reason="Too many requests",
+                ),
+            ),
+            repeat=True,
+        )
+
+        with pytest.raises(ClientResponseError, match="Too many requests"):
+            client.download(storage, [Pointer(oid=FOO_OID, size=FOO_SIZE)])
+
+        results = recorder[lfs_batch_url]
```

Comment:
can we simplify the test and use `lfs_batch_url` to test it, instead of `lfs_object_url`?

Reply:
I'm not sure I understand your proposal. Here, I'd like to make sure that rate limit retries work for the object API.

Reply:
yes, I see the difference now